[mod] manage: add themes.live command (rebuild on modification)

return42 opened the pull request:
## What does this PR do?

[mod] manage: add themes.live command (rebuild on modification)

## Why is this change important?

To get live builds while editing the HTML client of a theme.  The first argument of the command is the theme name:

    ./manage themes.live simple

To get live builds on `make run` use environment variable:

    LIVE_THEME=simple make run

## How to test this PR locally?

Start developer session:

    LIVE_THEME=simple make run

and edit CSS (LESS) & JS of the simple theme.

## Related issues

- https://github.com/searxng/searxng/pull/473#discussion_r745371694
- https://github.com/searxng/searxng/issues/498#issuecomment-1002286730

return42 committed:
[mod] manage: add themes.live command (rebuild on modification)

To have a live build while editing the HTML client of a theme.  The first
argument of the command is the theme name:

    ./manage themes.live simple

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[fix] remove leftover from 59f4c792

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

not-my-profile commented on manage:
```diff
@@ -698,14 +712,32 @@ themes.all() {
+            theme=searx/static/themes/$1
+            ;;
+        *)
+            die 42 "missing theme argument"
```
That error message also comes when misspelling a theme name in which case "missing argument" is misleading.

Also in the case of an error the error message is printed but since `themes.live` runs in a background job the `die` doesn't terminate the server ... which is a bit suboptimal since developers might miss the error message because with the default DEBUG startup output it will probably scroll of screen.

return42 replied:
@not-my-profile thanks for your helping remarks ..

> That error message also comes when misspelling a theme name

I changed the argument handling ...

```
$ ./manage themes.live xyz
ERROR: ./manage: line 792: themes.live(): unknown theme 'xyz' // [simple|oscar]

$ ./manage themes.live 
ERROR: ./manage: line 792: themes.live(): missing theme argument
```

> Also in the case of an error the error message is printed but since `themes.live` runs in a background job the `die` doesn't terminate the server 

I rearranged the processes and kill the process group if `themes.live` dies

```
$ LIVE_THEME=xyz make run
INFO    searx                         : load the default settings from /x/share/searxng/searx/settings.yml
INFO    searx                         : max_request_timeout=None
...
ERROR: ./manage: line 124: themes.live(): unknown theme 'xyz' // [simple|oscar]
make: *** [Makefile:28: run] Terminated
$
```

return42 replied:
@not-my-profile is the issue you named solved by my solution https://github.com/searxng/searxng/pull/664#discussion_r776499181?

not-my-profile replied:
Yes, thanks!

return42 committed:
[mod] live build of a theme: LIVE_THEME=simple make run

Environment variable to get live builds while modifying CSS & JS of a theme::

    LIVE_THEME=simple make run

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>
